atom/browser/api/atom_api_web_contents.cc

```diff
@@ -1433,18 +1433,30 @@ void WebContents::DevToolsClosed() {
   Emit("devtools-closed");
 }
 
+struct WebContents::DispatchHelper {
+  WebContents* web_contents;
+  content::RenderViewHost* render_view_host;
+  bool Send(IPC::Message* msg) { return render_view_host->Send(msg); }
+  void OnRendererMessageSync(const base::string16& channel,
+                             const base::ListValue& args,
+                             IPC::Message* message) {
+    web_contents->OnRendererMessageSync(channel, args, message);
+  }
+};
+
 bool WebContents::OnMessageReceived(const IPC::Message& message) {
   if (is_being_destroyed_)
     return false;
 
   bool handled = true;
+  DispatchHelper helper = {this, web_contents()->GetRenderViewHost()};
```

Comment:
This will not correctly route for OOPIFs. It should use the version of `OnMessageReceived` that takes a RenderFrameHost. I'm fixing right now

Reply:
actually there are even more changes to make for the js IPC api to properly handle `event.sender`